Tests/Help/Help.Tests.ps1

```diff
@@ -0,0 +1,64 @@
+﻿[String]$ModuleName = "Lability"
+[String]$ModuleManifestName = "{0}.psd1" -f $ModuleName
+
+Import-Module $PSScriptRoot\..\..\$ModuleManifestName -Force
+
+#Section mostly based on the blog post http://www.lazywinadmin.com/2016/05/using-pester-to-test-your-comment-based.html
+#Author: François-Xavier Cat fxcat[at]lazywinadmin[dot]com
+#Corrected by Wojciech Sciesinski wojciech[at]sciesinski[dot]net
+
+Describe "Module $ModuleName functions helps" -Tags "Help" {
+    
+    $FunctionsList = (get-command -Module $ModuleName | Where-Object -FilterScript { $_.CommandType -eq 'Function'} ).Name
+    
+    ForEach ($Function in $FunctionsList)
+    {
+        # Retrieve the Help of the function
+        $Help = Get-Help -Name $Function -Full
+        
+        #Parsing Notes can be added also
+        #$Notes = ($Help.alertSet.alert.text -split '\n')
+		
+		$Links = ($Help.relatedlinks.navigationlink.uri -split '\n')
+        
+        # Parse the function using AST
+        $AST = [System.Management.Automation.Language.Parser]::ParseInput((Get-Content function:$Function), [ref]$null, [ref]$null)
+		
+        Context "$Function - Help"{
+            
+            It "Synopsis"{ $help.Synopsis | Should not BeNullOrEmpty }
+            It "Description"{ $help.Description | Should not BeNullOrEmpty }
+            			
+            # Get the parameters declared in the Comment Based Help
+            $RiskMitigationParameters = 'Whatif', 'Confirm'
+            [String[]]$HelpParameters = $help.parameters.parameter | Where-Object name -NotIn $RiskMitigationParameters
+			
+            # Get the parameters declared in the AST PARAM() Block
+            [String[]]$ASTParameters = $AST.ParamBlock.Parameters.Name.variablepath.userpath
+			            
+            It "Parameter - Compare amount of parameters Help vs AST" {
+                $HelpParameters.count -eq $ASTParameters.count | Should Be $true
+            }
+            
+            # Parameter Description
+            $help.parameters.parameter | ForEach-Object {
```

Comment:
@iainbrighton - this is where the issue lies as it should really be

```powershell
$HelpParameters | Foreach-Object
```

However I've got a mitigation function for this called Get-CommonParameter in the PesterHelpers Module
https://github.com/PowerShellModules/PesterHelpers/blob/master/Private/Get/Get-CommonParameter.ps1

I then call it in the New-FunctionPesterTest function and basically remove all common parameters from the list of parameters to check on for help and other key information (in latest build being pushed shortly)